src/Microsoft.DotNet.Build.Tasks/RemoveDuplicatesWithLastOneWinsPolicy.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Given a list of items, remove duplicate items. Metadata is not considered. Case insensitive.
+    /// Unlike <see cref="RemoveDuplicates"/>, the last item in the list wins rather than the first.
+    /// That distinction is important when the items have different metadata.
+    /// </summary>
+    public class RemoveDuplicatesWithLastOneWinsPolicy : Task
+    {
+        /// <summary>
+        /// The list of items from which to remove duplicates.
+        /// </summary>
+        public ITaskItem[] Inputs
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// The list of items with duplicates removed.
+        /// </summary>
+        [Output]
+        public ITaskItem[] Filtered
+        {
+            get;
+            set;
+        }
+
+        public override bool Execute()
+        {
+            var existingIndexMap = new Dictionary<string, int>(Inputs.Length, StringComparer.OrdinalIgnoreCase);
+            var filteredList = new List<ITaskItem>(Inputs.Length);
+
+            foreach (ITaskItem item in Inputs)
+            {
+                int existingIndex;
+                if (existingIndexMap.TryGetValue(item.ItemSpec, out existingIndex))
+                {
+                    filteredList[existingIndex] = item;
+                }
+                else
+                {
+                    filteredList.Add(item);
+                    existingIndexMap.Add(item.ItemSpec, filteredList.Count - 1);
+                }
+            }
+
+            Filtered = filteredList.ToArray();
+            return true;
```

Comment:
Interesting. It looks like me like Distinct picks the first match and not the last one. I assume DistinctBy would do the same. 

https://github.com/dotnet/corefx/blob/872bb70953bcfe0a47dcc6e9789fbd330e295bc3/src/System.Linq/src/System/Linq/Enumerable.cs#L924

Mind you, I suppose one could reverse the input beforehand.